### PR TITLE
kymo: Fix erroneous slicing of kymos

### PR DESCRIPTION
**Pull request by tobiasjj**
**Why**
While following and rewriting fixtures to test kymo slicing, I stumbled over a weird corner case when slicing Kymos:

Kymo_slicing_error

**Explanation**
Slicing by providing only the start time with a value greater than the start timestamp of the very last line and less than or equal to the very last timestamp of the infowave created a dysfunctional Kymo. To fix it, the method to calculate the line timestamps in `Kymo.__getitem__()` was replaced with the method `Kymo.line_timestamp_ranges()`. Additionally, a check was introduced to ensure the stop timestamp of the kymo to have a value greater or equal to the stop timestamp of the last Kymo line, if the stop value of the slice item was greater than the start timestamp of the last Kymo line. As a bonus the old functions to calculate line and image timestamps are not needed anymore and were removed.

~**Why still a draft**
While rewriting the test for slicing, probably I found another bug in slicing. Investigating and working on it to be fixed ...~ It was an error in the tests.

### CODE REVIEW

**Review by JoepVanlier — CHANGES_REQUESTED** [review]
Good catch, but could you add a regression test for this?

File: lumicks/pylake/kymo.py [hunk 1]
```diff
@@ -212,16 +222,6 @@ def _tiff_timestamp_ranges(self) -> list:
         ts_ranges = np.array(self.line_timestamp_ranges())
         return [(np.min(ts_ranges), np.max(ts_ranges))]
 
-    @cachetools.cachedmethod(lambda self: self._cache)
```
[the remark this answers is not in the document]
Comment (tobiasjj) [reply lumicks/pylake/kymo.py:215]: > Good catch, but could you add a regression test for this?

I would like to add a regression test, however, the way the current tests are setup it is not easily possible. The test kymo used for slicing tests does not have padded pixel lines and in turn, the last timestamp of the infowave is less than the last stop timestamp of the last line. Therefore, this specific bug can not be triggered with that specific test data set. I integrated this regression test in another upcoming PR that I'm preparing right now ...

**Comment by tobiasjj** [comment]
> Good catch, but could you add a regression test for this?

I would like to add a regression test, however, the way the current tests are setup it is not easily possible. The test kymo used for slicing tests does not have padded pixel lines and in turn, the last timestamp of the infowave is less than the last stop timestamp of the last line. Therefore, this specific bug can not be triggered with that specific test data set. I integrated this regression test in another upcoming PR that I'm preparing right now ...

File: changelog.md [hunk 2]
```diff
@@ -31,6 +31,7 @@
 * Computing diffusion constants from temporally downsampled kymographs is now explicitly disallowed.
 * Fixed a bug where the `pixel_threshold` could be set to zero for an empty image. Now the minimum `pixel_threshold` is one.
 * Fixed a bug where single pixel detections in a `KymoTrackGroup` would contribute values with a dwell time of zero. These are now dropped, the correct minimally observable time is set appropriately and a warning is issued.
+* Fixed slicing of a `Kymo` where only providing the start time with a value greater than the start timestamp of the very last line and less than or equal to the very last timestamp of the infowave created a dysfunctional `Kymo`.
```
[the remark this answers is not in the document]
Comment (tobiasjj) [reply changelog.md:34]: Much better!

File: lumicks/pylake/tests/test_imaging_confocal/conftest.py [hunk 3]
```diff
@@ -22,6 +22,32 @@ def reference_counts():
...
+          [0., 0., 0.],
+          [0., 0., 0.],
+          [1., 1., 1.]],
+
+         [[2., 2., 2.],
+          [0., 0., 0.],
+          [1., 1., 1.],
+          [0., 0., 0.]]]
+    )
 # fmt: on
```
Comment (JoepVanlier) [inline lumicks/pylake/tests/test_imaging_confocal/conftest.py:51]: Is real content really needed? Wouldn't a simple `np.ones` or `np.eye` suffice for testing this bug? Since you don't need to check against any content it just seems like adding a lot of lines you don't actually use.

**Review by JoepVanlier — APPROVED** [review]
Looks good. One minor request, but then it's good to go I think.

File: lumicks/pylake/tests/test_imaging_confocal/conftest.py (hunk 3 above)
Comment (tobiasjj) [reply lumicks/pylake/tests/test_imaging_confocal/conftest.py]: You are right, this is not necessary. Replaced the `reference_image` with `np.ones` with the same shape.
